### DISCUSSION TITLE
Getting 500 Server Error trying to load historical data

### DISCUSSION THREAD

**Discussion by HyruleWarrior** [jertel/vuegraf#133] [Q&A] [answered]
I am trying to start fresh with `influxdb2` and so after configuring everything I am running a `--resetdatabase --historydays=720` command to pull in my historical data into new setup. This seemed to work ok the first time I tried it, but my dashboard was missing a lot of data I think due to how my device was named (didn't have "Panel" in the name). I renamed my device in Emporia app to `Main Panel` and updated my `vuegraf.json` config to match it:
```
"accounts": [
        {
            "name": "Main Panel",
            "email": "<email>",
            "password": "<password>",
            "devices": [
                {
                    "name": "Main Panel",
                    "channels": [
                        "Air Conditioner",
                        "Dryer",
                        "Dishwasher",
                        "Heat",
                        "Kitchen",
                        "HVAC Fan",
                        "EV Charger",
                        "Garage",
                        "Hot Tub",
                        "Back Deck",
                        "Living Room",
                        "Office",
                        "Insta Hot",
                        "Water Heater",
                        "Range",
                        "Oven"
                    ]
                }
            ]
        }
    ]
```

But now when I try resetting database and pulling in historical data I am getting 500 error:

```
2023-09-07 18:30:56.331632 | INFO  | Loading historical data: 720 day(s) ago
2023-09-07 18:32:27.715876 | ERROR | Failed to record new usage data: (<class 'requests.exceptions.HTTPError'>, HTTPError('500 Server Error: Internal Server Error for url: https://api.emporiaenergy.com/AppAPI?apiMethod=getChartUsage&deviceGid=147378&channel=4&start=2022-02-11T00:00:00Z&end=2022-03-03T23:59:59Z&scale=1D&energyUnit=KilowattHours'), <traceback object at 0x7fbea266d9c0>)
Traceback (most recent call last):
  File "/opt/vuegraf/vuegraf.py", line 363, in <module>
    extractDataPoints(device, usageDataPoints, 'History', historyStartTime, historyEndTime)
  File "/opt/vuegraf/vuegraf.py", line 182, in extractDataPoints
    usage, usage_start_time = account['vue'].get_chart_usage(chan, historyStartTime, historyEndTime, scale=Scale.DAY.value, unit=Unit.KWH.value)
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pyemvue/pyemvue.py", line 109, in get_chart_usage
    response.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://api.emporiaenergy.com/AppAPI?apiMethod=getChartUsage&deviceGid=147378&channel=4&start=2022-02-11T00:00:00Z&end=2022-03-03T23:59:59Z&scale=1D&energyUnit=KilowattHours
```

Did I mess something up with renaming? I am never sure in Emporia app what is the "Account Name" vs "Device Name". I renamed the device I had and I assume that also changes the account name since I only have that one device?

**Comment by jertel** [comment, marked as answer]
It looks like Emporia's server is having trouble, based on that error. I don't see anything wrong with your configuration, but you could try reducing the history duration from 720 to 30, just to see if their server can handle that. It's also possible that they had an internal issue of their own during your history load.

As far as Account Name vs Device Name, the Account Name is primarily for distinguishing between multiple accounts. For example, suppose you are also collecting data for your family-owned restaurant. You can have both of those metrics all placed in the same InfluxDB DB, and then use the Account drop down list in the dashboard to choose which one to view:

![image](https://github.com/jertel/vuegraf/assets/16639611/ca3688f6-861c-45c8-82a7-cf7ecc6b647d)

Continuing with this example, you could name your home account "My Crib" and your business account "Bob's Burger Shop"

The Device Name is the one that must match the "Main Panel" name in your Emporia App.

**Reply by HyruleWarrior** [reply]
Yep, looks like it was a transient server issue on Emporia side. Starting container again worked fine this time. It was just unfortunate coincidence it happened after modifying config.

>As far as Account Name vs Device Name, the Account Name is primarily for distinguishing between multiple accounts.

Ah! So this is something specified by Vuegraft. I don't know why I assumed it was something tied to Emporia account, but that makes more sense.
